src/movement_controls.py

```diff
@@ -0,0 +1,316 @@
+    current_acceleration: float
+
+    # which control scheme to use
+    vehicle_type: int
+    # for switching between vehicle types
+    toggle: bool
+
+    def __init__(self):
+        self.current_velocity_x = 0
+        self.current_velocity_y = 0
+        self.current_turnVelocity = 0
+        self.current_acceleration = 0
+        self.acceleration_rate = 0.5
+        self.decceleration_rate = 0.3
+        self.brake_rate = 0.3
+        self.friction = 0.1
+        self.drive_speed = 100
+        self.turn_speed = 100
+        self.toggle = False
+        self.vehicle_type = 1
+
+    #   Drive input is called by the player and passed that player's input to set what the vehicle's
+    #    velocity and rotation should be
+    #   Does not change the vehicles position
+    def drive_input(self, delta_time, input, vehicle):
+        # for testing to change between driving types
+        # TODO:find a permanent place for this logic that is not in movement
+        if self.toggle:
+            if input.secondary_fire_button.value == False:
+                self.toggle = False
+        else:
+            if input.secondary_fire_button.value == True:
+                self.toggle = True
+                self.change_vehicle()
+
+        # updates the vehicles intended  velocity and rotation based on type
+        if self.vehicle_type == 0:
+            self.car(delta_time, input, vehicle)
+        elif self.vehicle_type == 1:
+            self.ghost(delta_time, input, vehicle)
+        elif self.vehicle_type == 2:
+            self.ghost_screen_facing(delta_time, input, vehicle)
+        elif self.vehicle_type == 3:
+            self.tank(delta_time, input, vehicle)
+        elif self.vehicle_type == 4:
+            self.mars_walker(delta_time, input, vehicle)
+
+    def change_vehicle(self):
+        self.current_acceleration = 0
+        self.current_velocity_x = 0
+        self.current_velocity_y = 0
+        self.current_turnVelocity = 0
+        self.vehicle_type += 1
+        if self.vehicle_type > 4:
+            self.vehicle_type = 0
+
+    # called from the player to tell the vehicle to act on it's intended velocity and rotation
+    def move(self, vehicle):
+        # find the X and Y movement based on the angle of the sprite
+        new_x_pos = self.current_velocity_x + vehicle.center_x
+        new_y_pos = self.current_velocity_y + vehicle.center_y
+
+        # NOTE: place holder boundaries
+        if new_x_pos < 0 or new_x_pos > 500 or new_y_pos < 0 or new_y_pos > 500:
+            # bounce of the imaginary collision and roll away
+            # vehicle.center_x -= self.current_velocity_x * 3
+            # vehicle.center_y -= self.current_velocity_y * 3
+            self.current_acceleration *= -0.1
+        else:
+            vehicle.center_x += self.current_velocity_x
+            vehicle.center_y += self.current_velocity_y
+
+        vehicle.angle += self.current_turnVelocity
+
+    def car(self, delta_time, input, vehicle):
```

Comment:
I don't know the term for this, but could you call out what variables are in your function inputs to make the code easier to the rest of us figure out what's going on and modify it in the future, or at least where they call out class instances?
For instance: def car(self, delta_time, input: PlayerInput, vehicle: ??):
I think vehicle is probably a sprite?

This applies to in many places but just putting the one comment here

Reply:
Non blocking I guess. This can be cleaned up later